src/core/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,233 @@
+ */
+export const TimePicker = (props: TimePickerProps): JSX.Element => {
+  const {
+    date,
+    onChange,
+    use12Hours = false,
+    precision = 'minutes',
+    hourStep = 1,
+    minuteStep = 1,
+    secondStep = 1,
+  } = props;
+
+  useTheme();
+
+  const [selectedTime, setSelectedTime] = React.useState(date);
+
+  React.useEffect(() => {
+    setSelectedTime(date);
+  }, [date]);
+
+  const onHourClick = (hour: number) => {
+    const adjustedSelectedDay = selectedTime ?? new Date();
+    adjustedSelectedDay.setHours(hour);
+    updateCurrentTime(adjustedSelectedDay);
+  };
+
+  const onMinutesClick = (minutes: number) => {
+    const adjustedSelectedDay = selectedTime ?? new Date();
+    adjustedSelectedDay.setMinutes(minutes);
+    updateCurrentTime(adjustedSelectedDay);
+  };
+
+  const onSecondsClick = (seconds: number) => {
+    const adjustedSelectedDay = selectedTime ?? new Date();
+    adjustedSelectedDay.setSeconds(seconds);
+    updateCurrentTime(adjustedSelectedDay);
+  };
+
+  const updateCurrentTime = (time: Date) => {
+    setSelectedTime(time);
+    onChange?.(time);
+  };
+
+  // const handleTimeKeyDown = (
+  //   event: React.KeyboardEvent<HTMLLIElement>,
+  //   maxValue: number,
+  //   callback: (value: number) => void,
+  //   onClick: (value: number) => void,
+  //   currentValue?: number,
+  // ) => {
+  //   if (!focusedTime || currentValue == null) {
+  //     return;
+  //   }
+  //   switch (event.key) {
+  //     case 'ArrowDown':
+  //       if (currentValue + 1 > maxValue) {
+  //         break;
+  //       }
+  //       callback(currentValue + 1);
+  //       needFocus.current = true;
+  //       event.preventDefault();
+  //       break;
+  //     case 'ArrowUp':
+  //       if (currentValue - 1 < 0) {
+  //         break;
+  //       }
+  //       callback(currentValue - 1);
+  //       needFocus.current = true;
+  //       event.preventDefault();
+  //       break;
+  //     case 'Enter':
+  //     case ' ':
+  //     case 'Spacebar':
+  //       onClick(currentValue);
+  //       event.preventDefault();
+  //       break;
+  //   }
+  // };
+
+  const scrollIntoView = (ref: HTMLLIElement | null, isSame: boolean) => {
+    isSame && ref?.scrollIntoView({ block: 'nearest', inline: 'nearest' });
+  };
+
+  const getTimeTemplate = (
+    data: number[],
+    step: number,
+    isSame: (number: number, date: Date | undefined) => boolean,
+    onClick: (number: number) => void,
+  ) => {
+    return (
+      <div className='iui-time'>
+        <ol>
+          {data
+            .filter((_, index) => index % step === 0)
+            .map((value) => {
+              return (
+                <li
+                  className={cx({
+                    'iui-selected': isSame(value, selectedTime),
+                  })}
+                  key={value}
+                  tabIndex={isSame(value, selectedTime) ? 0 : undefined}
```

Comment:
Yeah, this won't be nice for keyboard navigation, better remove, even if you're not gonna add arrow key support initially.

When it comes to accessibility, "none" is better than "bad".

Reply:
Check up current state, if its "bad".